src/Helper/Url.php

```diff
@@ -103,7 +103,57 @@ public function __invoke($name = null, $params = [], $options = [], $reuseMatche
 
         $options['name'] = $name;
 
-        return $this->router->assemble($params, $options);
+        try {
+            return $this->router->assemble($params, $options);
+        } catch (\Zend\Router\Exception\InvalidArgumentException $e) {
+            throw new Exception\InvalidArgumentException(
```

Comment:
this is duplicated now 4x, we can extract it to separate private method or we can have it after try...catch, something like:
```php
...
} catch (Exception1 $e) {
} catch (Exception2 $e) {
}

throw new Exception\InvalidArgumentException(
    sprintf(
        'Couldnt create URL for route "%s", params "%s" and options "%s": %s',
         $name,
         json_encode($params),
         json_encode($options),
         $e->getMessage()
    ),
     $e->getCode(),
     $e
);
```

And we need test case for ALL of these exception types. Not just for one.